src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+import org.w3c.dom.Node;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private float threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int hash = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (table[hash] == null) {
+            table[hash] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = table[hash];
+        while (currentNode.next != null && !Objects.equals(key,currentNode.key)) {
+            currentNode = currentNode.next;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            currentNode.value = value;
+            return;
+        }
```

Comment:
rewrite so that you don't check equality twice

Reply:
not fixed